kafka_utils/kafka_check/status_code.py

```diff
@@ -26,14 +31,19 @@
 }
 
 
-def prepare_terminate_message(string):
+class TerminateMessageDict(TypedDict):
+    message: str
+    raw: str
+
+
+def prepare_terminate_message(string: str) -> TerminateMessageDict:
```

Comment:
It seems like every time we use this function we cast to a `dict[str, Any]` anyway? What's the general approach you are using to decide when to do stricter typing (e.g. `TypedDict` stuff) vs simpler/less code (e.g. just mark the return type as `-> dict` or something)? Or is it all just generated by some tool?

Reply:
That's a good question and I don't have a perfect answer. The reason I used `TerminateMessageDict` in this case is that it has a concrete type. However, the usages of it were more general. Maybe it's not worth it in practice, but I prefer stricter types like this where possible. In theory it makes refactoring this to code to be better designed from a type safety perspective easier.